ananse/influence.py

```diff
@@ -24,31 +25,58 @@
 Expression = namedtuple("Expression", ["score", "absfc", "realfc"])
 
 
-def read_top_interactions(fname, edges=100_000, sort_by="prob"):
+def read_top_interactions(
+    source, target, edges: Union[int, None] = 100_000, sort_by="prob"
+):
     """
-    Read network file and return the top interactions.
+    Read two network files and return the top source to target interactions,
+    using the designated scoring metric (by default the combined probability).
 
-    Incorporates the combined metric (probability score) by default to select the most
-    certain interactions, however other sorting options such as the weighted binding score
-    can be applied too.
+    If edges is none, return all shared interactions.
     """
-    # read the GRN file
-    rnet = pd.read_csv(
-        fname,
-        usecols=["tf_target", sort_by],
-        sep="\t",
-        dtype={"tf_target": str, sort_by: "float32"},
+    # filter for shared edges
+    def read_minimal_network(fname, col):
+        return pd.read_csv(
+            fname,
+            usecols=["tf_target", col],
+            sep="\t",
+            dtype={"tf_target": str, col: "float32"},
+            index_col="tf_target",
+        )
+
+    source_int = read_minimal_network(source, sort_by)
+    source_int.rename(columns={sort_by: "source"}, inplace=True)
+    target_int = read_minimal_network(target, sort_by)
+    target_int.rename(columns={sort_by: "target"}, inplace=True)
+
+    top_int = pd.merge(
+        target_int,
+        source_int,
+        left_index=True,
+        right_index=True,
+        how="inner",  # shared
```

Comment:
If I understand correctly, this is not correct. What we want are all interactions that are relevant from the *target* network. This can mean two things:
1) An interaction that is *only* present in the target network.
2) An interaction that is *also* present in the source network, but *higher* in the target network.

Here is an example. 

Source network
```
A    B    1
A    C    0.5
B    C    0.1
````

Target network
```
A    B    0.9
A    C    0.8
B    C    0.6
C    D    0.1
```

In this case, the differential network would be:
```
B    C    0.5
A    C    0.3
C    D    0.1
```

It is now sorted by differential weight/probability, which can be used to select the top edges.